backend/cn/solver.mli

```diff
@@ -1,25 +1,36 @@
+(* Module Solver -- Interface to the SMT solver via SMTLIB *)
 type solver
 type model
+(* (TODO: BCP: The "with quantifiers" part will be the instantiations that the
+   solver found -- is that right?) *)
```

Comment:
Yes. To implement `provable (Forall ((s,bt), t))`, CN picks a fresh variable `s'` and checks that `not t[s'/s]` is unsatisfiable ("there exists no choice of s such that t doesn't hold"). If `provable` returns false, i.e. the SMT solver found an `s'` for which `t[s'/s]` doesn't hold, then `model_with_q` has that `s'` for its second component.